aiorospy/src/aiorospy/action.py

```diff
@@ -196,42 +234,66 @@ def send_goal(self, goal):
 class AsyncActionServer:
     """ Async wrapper around the action server API. """
 
-    def __init__(self, name, action_spec, coro, loop=None):
+    def __init__(self, name, action_spec, coro, loop=None, ):
         """ Initialize an action server. Incoming goals will be processed via the speficied coroutine. """
         self.name = name
         self._loop = loop if loop is not None else asyncio.get_event_loop()
         self._coro = coro
         self._tasks = {}
 
+        self._exc_q = janus.Queue(loop=loop)
+        self._goal_q = janus.Queue(loop=loop)
+        self._cancel_q = janus.Queue(loop=loop)
+
         self._server = ActionServer(
             name, action_spec, goal_cb=self._goal_cb, cancel_cb=self._cancel_cb, auto_start=False)
 
+    async def start(self):
+        # TODO(pbovbel) actionservers don't stop and cause the program to not terminate
```

Comment:
This is an old frustration with python actionservers that's rearing it's ugly head. We TODO need to patch in some sort of server shutdown method, otherwise the process will never come down.